samcli/hook_packages/terraform/hooks/prepare.py

```diff
@@ -905,14 +908,171 @@ def _generate_makefile_rule_for_lambda_resource(
+        _build_show_command_rule(
+            python_command_name, output_dir, resource_address, sam_metadata_resource, terraform_application_dir
+        )
+    )
+    return f"{target}{apply_command}{show_command}"
+
+
+def _build_show_command_rule(
+    python_command_name: str,
+    output_dir: str,
+    resource_address: str,
+    sam_metadata_resource: SamMetadataResource,
+    terraform_application_dir: str,
+) -> str:
+    """
+    Build the Terraform show command to be used inside of the Makefile rule
+
+    Parameters
+    ----------
+    python_command_name: str
+        the python command name to use for running a script in the makefile rule
+    output_dir: str
+        the directory into which the Makefile is written
+    resource_address: str
+        Address of a given terraform resource
+    sam_metadata_resource: SamMetadataResource
+        A sam metadata resource; the generated makefile rule will correspond to building the lambda resource
+        associated with this sam metadata resource
+    terraform_application_dir: str
+        the terraform project root directory
+
+    Returns
+    -------
+    str
+        Fully resolved Terraform show command
+    """
+    show_command_template = (
+        "terraform show -json | {python_command_name} {terraform_built_artifacts_script_path} "
+        "{jpath_string} $(ARTIFACTS_DIR) {project_root_dir}"
+    )
+    jpath_string = _build_jpath_string(sam_metadata_resource, resource_address)
+    terraform_built_artifacts_script_path = Path(output_dir, TERRAFORM_BUILD_SCRIPT).relative_to(
+        terraform_application_dir
+    )
+    return show_command_template.format(
+        python_command_name=python_command_name,
+        terraform_built_artifacts_script_path=terraform_built_artifacts_script_path,
+        jpath_string=jpath_string,
+        project_root_dir=terraform_application_dir,
+    )
+
+
+def _build_jpath_string(sam_metadata_resource: SamMetadataResource, resource_address: str) -> str:
+    """
+    Constructs the JPath string for a given sam metadata resource from the planned_values
+    to the build_output_path as is created by the Terraform plan output
+
+    Parameters
+    ----------
+    sam_metadata_resource: SamMetadataResource
+       Full address of a Terraform resource
+
+    resource_address: str
+        Full address of a Terraform resource
+
+    Returns
+    -------
+    str
+       Full JPath string for a resource from planned_values to build_output_path
+    """
+    jpath_string_template = (
+        "|planned_values|root_module{child_modules}|resources"
+        '[?address=="{resource_address}"]|values|triggers|built_output_path'
+    )
+    child_modules_template = "|child_modules[?address=={module_address}]"
+    module_address = sam_metadata_resource.current_module_address
+    full_module_path = ""
+    if module_address:
+        parent_modules = _get_parent_modules(module_address)
+        if parent_modules:
```

Comment:
Good idea. The `if module_address` needs to be checked though since `sam_metadata_resource.current_module_address` is defined as `Optional` otherwise the linter won't like it.

Reply:
you can make the parameter optional in `_get_parent_modules` to make the linter happy lol.
You already check `if not module_address` anyway